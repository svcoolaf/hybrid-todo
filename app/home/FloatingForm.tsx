import { useMutation } from "@apollo/react-hooks";
import React, { useState } from "react";
import styled from "styled-components";
import theme from "../common/lib/theme";
import { Todo } from "../common/models/Todo";
import Icon from "./Icon";

const Form = styled.form`
  line-height: ${theme.sizes.form}px;
  width: 100vw;
  display: flex;
  align-items: center;
  background: ${theme.colors.white};
  padding-left: ${theme.sizes.quadruple}px;
`;

const Input = styled.input`
  margin: 0 0 0 ${theme.sizes.double}px;
  padding: ${theme.sizes.double}px;
  border: 0;
  flex-grow: 1;
  line-height: ${theme.sizes.form}px;
`;

const Floating = styled.div`
  position: fixed;
  bottom: 0;
  border-top: 1px ${theme.colors.light} solid;
`;

const Clickable = styled.button`
  cursor: pointer;
  padding: 0;
  border: 0;
  background: transparent;
`;

const FloatingForm = ({ triggerFetch }) => {
  const [createJob] = useMutation<any, Partial<Todo>>(Todo.create);
  const [content, setContent] = useState("");
  const onFormSubmit = e => {
    e.preventDefault();
    if (!content) {
      return;
    }
    createJob({
      variables: { content }
    }).then(() => {
      setContent("");
      triggerFetch();
    });
  };
  return (
    <Floating>
      <Form onSubmit={onFormSubmit}>
        <Clickable onClick={onFormSubmit}>
          <Icon name="plus" />
        </Clickable>
        <Input
          value={content}
          onChange={e => {
            setContent(e.target.value);
          }}
          placeholder="Add todo"
        />
      </Form>
    </Floating>
  );
};

export default FloatingForm;
