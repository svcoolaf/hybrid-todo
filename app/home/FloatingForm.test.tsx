import { noop } from "lodash";
import React from "react";
import renderer from "react-test-renderer";
import FloatingForm from "./FloatingForm";

describe("FloatingForm", () => {
  it("renders", () => {
    const tree = renderer.create(<FloatingForm triggerFetch={noop} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
