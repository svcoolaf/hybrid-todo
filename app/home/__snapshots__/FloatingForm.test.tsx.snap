// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FloatingForm renders 1`] = `
<div
  className="FloatingForm__Floating-sc-6zf4dp-2 ggdTjX"
>
  <form
    className="FloatingForm__Form-sc-6zf4dp-0 bVzonV"
    onSubmit={[Function]}
  >
    <button
      className="FloatingForm__Clickable-sc-6zf4dp-3 evHQIf"
      onClick={[Function]}
    >
      <img
        alt="plus"
        height={24}
        src="/static/icons/plus.svg"
        width={24}
      />
    </button>
    <input
      className="FloatingForm__Input-sc-6zf4dp-1 eolwPR"
      onChange={[Function]}
      placeholder="Add todo"
      value=""
    />
  </form>
</div>
`;
